pgarchive

```diff
@@ -0,0 +1,1039 @@
+# You also need to enable hot_standby in postgresql.conf for this.
+pause_at_recovery_target = false
+
+EOF
+
+    sed -i".snapshot" '/@@@standby-config-start@@@/,/@@standby-config-end@@@/d' "$clone/postgresql.conf"
+    cat >>"$clone/postgresql.conf" <<EOF
+
+# @@@clone-config-start@@@
+# default overrides for a clone, feel free to change them as needed
+listen_addresses = ''
+port = $next_clone_port
+shared_buffers = 2GB
+checkpoint_segments = 16
+synchronous_standby_names = ''
+wal_keep_segments = 0
+#autovacuum = off
+#fsync = off
+#synchronous_commit = off
+# @@@clone-config-end@@@
+
+EOF
+
+    incr_next_clone_port
+}
+
+clone_duplicate() {
+
+    dup="$clones/$1"
+    [[ -n "$1" ]] || fail "new clone name required"
+    [[ -e "$dup" ]] && fail "clone $1 already exists"
+
+    btrfs subvolume snapshot "$clone" "$dup"
+    rm -f "$dup"/postmaster.*
+    log "duplicated from clone $(basename $clone)" >> "$dup/pgarchive.log"
+
+    echo "# override port for duplicate" >>"$clone/postgresql.conf"
+    echo "port = $next_clone_port" >>"$clone/postgresql.conf"
+    echo "" >>"$clone/postgresql.conf"
+
+    incr_next_clone_port
+}
+
+
+### main
+
+cmd=$1
+shift || true
+
+case "$cmd" in
+    help|--help)
+        help
+        ;;
+
+    standby-restore)
+        standby_restore $@
+        ;;
+
+    standby-record-cleanup-position)
+        standby_record_cleanup_position $@
+        ;;
+
+    clone-restore)
+        clone_restore $@
+        ;;
+
+    bash-completion)
+        bash_completion
+        ;;
+
+    container)
+        subcmd=$1
+        shift || true
+
+        case "$subcmd" in
+            init)
+                # no configure here, as we don't have anything yet
+                set_container
+                setup
+                container_init
+                ;;
+
+            purge)
+                # no configure here either, configuration may be present or not
+                set_container
+                setup
+                [[ -f $config ]] && . $config
+
+                $0 wal_archive status >/dev/null 2>&1 && fail "wal_archive must be stopped"
+                $0 standby status >/dev/null 2>&1 && fail "standby must be stopped"
+                ($0 clone list --status | grep "server is running") >/dev/null 2>&1 && fail "all clones must be stopped"
+
+                if [[ "$1" = '--force' ]]; then
+                    log "Completely purging $container."
+                    log "Any errors are ignored and left to manual cleanup, or you may repeat this."
+                    container_purge
+                else
+                    log "If you really want to purge $container and remove upstream's $slot slot, provide --force."
+                    log "WARNING: this will delete EVERTHING under $container."
+                fi
+                ;;
+
+            disconnect)
+                # remove the replication slot & ?
+                fail "TODO"
```

Comment:
Yes, disconnect & resync are future features. We'll move to using issues soon, until then bear with me :-)